libmamba/src/core/util_string.cpp

```diff
@@ -7,15 +7,117 @@
 #include "mamba/core/util_string.hpp"
 
 #include <cctype>
+#include <cwchar>
 #include <cwctype>
-#include <exception>
-#include <type_traits>
+#include <stdexcept>
+
+#include <stddef.h>
 
 namespace mamba
 {
-    /***************************************************
-     *  Implementation of to_lower to_upper functions  *
-     ***************************************************/
+    /****************************************
+     *  Implementation of cctype functions  *
+     ****************************************/
+
+    bool is_control(char c)
+    {
+        return std::iscntrl(static_cast<unsigned char>(c)) != 0;
+    }
+
+    bool is_control(wchar_t c)
+    {
+        return std::iswcntrl(static_cast<wint_t>(c)) != 0;
```

Comment:
According to the [documentation](https://en.cppreference.com/w/cpp/string/wide/iswcntrl), I don't think that we need the `static_cast` to call this function. It also makes it inconsistent with the cast in the equivalent function above which uses `char` and that we cast to a type different from the one specified in the definition of the function, for the sake of a defined behavior (same applies for the other similar functions below).